Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Search/Filter.cpp

```diff
@@ -404,6 +404,30 @@ namespace AzToolsFramework
             return entry->GetEntryType() == m_entryType;
         }
 
+        //////////////////////////////////////////////////////////////////////////
+        // AssetPathFilter
+        //////////////////////////////////////////////////////////////////////////
+        AssetPathFilter::AssetPathFilter()
+        {
+        }
+
+        void AssetPathFilter::SetAssetPath(AZ::IO::Path path)
+        {
+            m_assetPath = QString::fromUtf8(path.c_str());
+        }
+
+        QString AssetPathFilter::GetNameInternal() const
+        {
+            return m_assetPath;
+        }
+
+        bool AssetPathFilter::MatchInternal(const AssetBrowserEntry* entry) const
+        {
+            QStringList projectDirectory = m_assetPath.split("\\");
+            QString entryPath = QString::fromUtf8(entry->GetFullPath().data());
+            return StringMatch(projectDirectory.last(), entryPath);
```

Comment:
The `StringMatch` method only checks if text contains what is being searched for, but we can be more explicit than that. We should instead check if the entryPath begins with the project directory path, which will ensure that the entry exists in a sub-directory of the intended search path. Instead of splitting by `\\` which would only work on windows, you can use the AZ::IO::Path to make sure the paths you are comparing have the same path separators.

There is even a `IsRelativeTo` on `AZ::IO::BasicPath` that looks like it might do the trick as well.

Reply:
Thanks for the feedback, I wasn't sure what the best way to filter for path descendants was. Fixed it with the `IsRelativeTo` method!